### PR TITLE
Upgrading async profiler version to 1.7.1 and adding safemode config

### PR DESCRIPTION
**Pull request by eyalkoren**
Adding support for the `safemode` config by async profiler through an internal agent config.

### CODE REVIEW

File: apm-agent-plugins/apm-profiling-plugin/src/main/java/co/elastic/apm/agent/profiler/ProfilingConfiguration.java [hunk 1]
```diff
@@ -60,6 +60,13 @@
         .tags("added[1.15.0]", "experimental")
         .buildWithDefault(false);
 
+    private final ConfigurationOption<Integer> recoverMode = ConfigurationOption.<Integer>integerOption()
+        .key("recover_mode")
+        .configurationCategory(PROFILING_CATEGORY)
+        .dynamic(false)
+        .tags("internal")
+        .buildWithDefault(0);
```
Comment (felixbarny) [inline apm-agent-plugins/apm-profiling-plugin/src/main/java/co/elastic/apm/agent/profiler/ProfilingConfiguration.java:68]: This disables all stack trace recoveries by default. Is that intentional?

File: apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/RecoverSamplingProfilerTest.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            .until(() -> profiler.getProfilingSessions() > 1);
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        tracer.stop();
+    }
+
+    @Test
+    void testAll() throws Exception {
```
Comment (felixbarny) [inline apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/RecoverSamplingProfilerTest.java:85]: What's the value of testing the different options?

File: apm-agent-plugins/apm-profiling-plugin/src/main/java/co/elastic/apm/agent/profiler/ProfilingConfiguration.java (hunk 1 above)
Comment (eyalkoren) [reply apm-agent-plugins/apm-profiling-plugin/src/main/java/co/elastic/apm/agent/profiler/ProfilingConfiguration.java:68]: This is not what I plan to push to master, this is only an analysis agent, where we want to start by testing that the recovery code is not the cause. Only if required, we will ask the user to set this to other values.
I did it as a PR just so I have the artefacts publicly available.

File: apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/RecoverSamplingProfilerTest.java (hunk 2 above)
Comment (eyalkoren) [reply apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/RecoverSamplingProfilerTest.java:85]: Only sanity check that enabling discrete parts of the recovery logic doesn't break anything. Again - not to be eventually merged.

**Comment by eyalkoren** [comment]
@felixbarny any idea how to test the new state?

**Comment by felixbarny** [comment]
I don’t think testing is required here. We just bass this through to async profiler. We shouldn’t test async profiler itself in this project.

**Pull request merged**